integration/common/openlineage/common/provider/dbt.py

```diff
@@ -633,9 +633,21 @@ def extract_namespace(self, profile: Dict) -> str:
             return "bigquery"
         elif profile['type'] == 'redshift':
             return f"redshift://{profile['host']}:{profile['port']}"
+        elif profile['type'] == 'spark':
+            if profile['method'] in ['thrift', 'odbc', 'http']:
+                port = ""
+                if profile['port']:
+                    port = f":{profile['port']}"
+                return f"spark://{profile['host']}" + port
+            else:
+                raise NotImplementedError(
+                    f"Connection method `{profile['method']}` is not "
+                    f"supported for spark adapter."
+                )
         else:
             raise NotImplementedError(
-                f"Only 'snowflake', 'bigquery', and 'redshift' adapters are supported right now. "
+                f"Only 'snowflake', 'bigquery', 'redshift' and "
```

Comment:
If we define an `Adaptor` enum, we can list the values and build the error string based on the list. This will avoid always (and maybe sometimes forgetting) to update the error message.

Reply:
Done.